src/qt/bitcoingui.cpp

```diff
@@ -113,6 +113,7 @@ BitcoinGUI::BitcoinGUI(interfaces::Node& node, const PlatformStyle *_platformSty
         connect(walletFrame, &WalletFrame::message, [this](const QString& title, const QString& message, unsigned int style) {
             this->message(title, message, style);
         });
+        connect(walletFrame, &WalletFrame::currentWalletSet, [this] { updateWalletStatus(); });
```

Comment:
Do we really need to wrap that call with lambda? I see that `updateWalletStatus` is already a slot, or am I missing something?

Reply:
No, we don't.

The small benefit is that my editor could easy find this call place of `updateWalletStatus()` function member. It fails to find it by pointer `&BitcoinGUI::updateWalletStatus`.

Are there any drawbacks of such an approach?

Reply:
The drawback is writing more verbose / unnecessary code, for no reason other than to appease your editor. If it's been questioned during review, it's just as likely someone will open a PR to remove it later on, citing the same reasoning, that, it's confusing / unnecessary. We really shouldn't be bloating our codebase just because your editor doesn't work.

Reply:
I strictly prefer [`QObject::connect(const QObject *sender, PointerToMemberFunction signal, Functor functor)`](https://doc.qt.io/qt-5/qobject.html#connect-4) overload over [`QObject::connect(const QObject *sender, PointerToMemberFunction signal, const QObject *receiver, PointerToMemberFunction method, Qt::ConnectionType type)`](https://doc.qt.io/qt-5/qobject.html#connect-3) one for the following reasons:

- the former states used signal-slot parameters explicitly, i.e., here it is clear that we use no signal parameters
- the former is less verbose, compare `[this] { updateWalletStatus(); }` vs `this, &BitcoinGUI::updateWalletStatus`

> We really shouldn't be bloating our codebase just because your editor doesn't work.

Absolutely agree. But this is not that case, no?

Btw, this--very helpful syntax--is already actively used in the code syntax. E.g. https://github.com/bitcoin/bitcoin/pull/14123 and other prs by @promag.

Reply:
Thanks for providing the actual reasoning; this should have been the first comment. I don't really have an opinion on the code, but was always going to leave the response above, regardless of the change, because the justification "it's better for my editor", is not a good one.

Reply:
I apologize for that.

Reply:
No worries. No need for an apology. Feel free to mark as resolved and merge etc.